docs/write-filter-definitions.md

```diff
@@ -191,49 +191,615 @@ To simplify, when you have multiple items, you can combine their values into a s
 ```json
 {
   "filter": {
-    "class": [
-      { "name": [ "connection", "general", "table_access" ] }
-    ]
+    "class": [{ "name": ["connection", "general", "table_access"] }]
   }
 }
 ```
-This array object is more concise. You should use it if you do not plan to add additional settings for individual classes.
 
+This array object is more concise. You should use it if you do not plan to add additional settings for individual classes.
 
 #### List of event and subclass options
 
-The table shows the available event classes and their subclasses:
+The available event classes and their subclasses are as follows:
 
-<style>
-  table {
-    width: 100%;
-    border-collapse: collapse;
+##### General
+
+| Class Name | Event Subclass      | Description                                                                  |
+| ---------- | ------------------- | ---------------------------------------------------------------------------- |
+| General    | GeneralError        | Represents an error event with details about the error code and context.     |
+| General    | GeneralCommand      | Represents a command execution event, detailing the command issued.          |
+| General    | GeneralQuery        | Represents a query execution event, detailing the query that was run.        |
+| General    | GeneralUser         | Represents an event related to a specific user, detailing user information.  |
+| General    | GeneralHost         | Represents an event related to a specific host, detailing host information.  |
+| General    | GeneralSQLCommand   | Represents an event related to a specific SQL command executed.              |
+| General    | GeneralExternalUser | Represents an event related to an external user interacting with the system. |
+| General    | GeneralIP           | Represents an event related to a specific IP address involved in the event.  |
+
+A JSON code sample for General class:
+
+```json
+{
+  "filter": {
```

Comment:
The rule definition also has very little in common with reality
Here is how, for instance filtering by event classes / subclasses is done
```json
{
  "filter": {
    "class": [
      {
        "name": "connection",
        "event": [
          { "name": "connect" },
          { "name": "disconnect" }
        ]
      },
      { "name": "general" },
      {
        "name": "table_access",
        "event": [
          { "name": "insert" },
          { "name": "delete" },
          { "name": "update" }
        ]
      }
    ]
  }
}
```

Reply:
And here, for instance, is an example of how we can filter by fields from the "general" event
```json
{
  "filter": {
    "class": {
      "name": "general",
      "event": {
        "name": "status",
        "log": {
          "or": [
            {
              "and": [
                { "field": { "name": "general_command.str",    "value": "Query" } },
                { "field": { "name": "general_command.length", "value": 5 } }
              ]
            },
            {
              "and": [
                { "field": { "name": "general_command.str",    "value": "Execute" } },
                { "field": { "name": "general_command.length", "value": 7 } }
              ]
            }
          ]
        }
      }
    }
  }
}
```